src/nnet3/nnet-common.cc

```diff
@@ -392,11 +392,12 @@ size_t IndexVectorHasher::operator () (
                             // skipping over more elements.  Setting n1 large or
                             // n2 to 1 would make the hasher consider all
                             // elements.
+  size_t len = index_vector.size();
   // all long-ish numbers appearing below are randomly chosen primes.
-  size_t ans = 1433 + 34949  * index_vector.size();
+  size_t ans = 1433 + 34949 * len;
   std::vector<Index>::const_iterator iter = index_vector.begin(),
       end = index_vector.end(), med = end;
-  if (med > iter + n1)
+  if (n1 <= len)
```

Comment:
.. also I believe this should be `if n1 < len`, not `<=`.

Reply:
updated again. Also checked indentation and spaces. With another question:

why when `med == end == iter + n1`, we would like to do `for (; iter != med; ++iter)`  instead of skip this? This will make the hash better?